src/amo/reducers/recommendations.js

```diff
@@ -0,0 +1,175 @@
+/* @flow */
+import invariant from 'invariant';
+
+import { createInternalAddon } from 'core/reducers/addons';
+import type { ExternalAddonType } from 'core/types/addons';
+
+export const ABORT_FETCH_RECOMMENDATIONS: 'ABORT_FETCH_RECOMMENDATIONS'
+  = 'ABORT_FETCH_RECOMMENDATIONS';
+export const FETCH_RECOMMENDATIONS: 'FETCH_RECOMMENDATIONS'
+  = 'FETCH_RECOMMENDATIONS';
+export const LOAD_RECOMMENDATIONS: 'LOAD_RECOMMENDATIONS'
+  = 'LOAD_RECOMMENDATIONS';
+
+export type Recommendations = {|
+  addons: Array<ExternalAddonType>,
+  fallbackReason: string,
+  loading: boolean,
+  outcome: string,
+|};
+
+export type RecommendationsState = {
```

Comment:
I think we could have a stricter type here.

Reply:
We were unable to make this an exact type because of https://github.com/facebook/flow/issues/2405, but I see that should be fixed in Flow 0.71, which is landing in https://github.com/mozilla/addons-frontend/pull/4917, so let's get that landed and then I think I can update this.